sys/boot/i386/libi386/biosdisk.c

```diff
@@ -484,6 +484,11 @@ bd_open(struct open_file *f, ...)
 		}
 		if (geli_taste(bios_read, &dskp,
 		    entry->part.end - entry->part.start) == 0) {
+			if (geli_havekey(&dskp) == 0) {
```

Comment:
Don't bother looking for a password in the environment if we already have a key derived